src/FileGetContents.php

```diff
@@ -31,36 +41,68 @@ public function __construct()
         $this->options = array_replace_recursive($this->options, $options);
     }
 
+    /**
+     * @param string              $filename Name of the file to read.
+     * @param resource|array|null $context A valid context resource created with `stream_context_create()`. If you don't
+     *                                     need to use a custom context, you can skip this parameter.
+     *
+     * @return bool|string
+     */
     public function get($filename, $context = null)
     {
         $context = $this->getStreamContext($filename);
-        self::setHttpHeaders($context);
+        $context = self::setHttpHeaders($context);
+
         $result = file_get_contents($filename, null, $context);
+
         self::setLastResponseHeaders($http_response_header);
 
         return $result;
     }
 
+    /**
+     * @param array $headers
+     *
+     * @final Since 1.1.0
+     */
     public static function setLastResponseHeaders($headers)
     {
         self::$lastResponseHeaders = $headers;
     }
 
+    /**
+     * @return array|null
+     *
+     * @final Since 1.1.0
+     */
     public static function getLastResponseHeaders()
     {
         return self::$lastResponseHeaders;
     }
-    
+
+    /**
+     * @param array $headers An indexed or associative array.
+     *
+     * @final since 1.1.0
+     */
     public static function setNextRequestHeaders(array $headers)
     {
         self::$nextRequestHeaders = $headers;
     }
 
+    /**
+     * @return bool
+     *
+     * @final since 1.1.0
+     */
     public static function hasNextRequestHeaders()
     {
-        return !empty(self::$nextRequestHeaders);
+        return null !== self::$nextRequestHeaders;
```

Comment:
Actually probably not acceptable for a minor version as technically it's a BC break: empty request headers would have returned `true` before.

Tell me if the change of behaviour is wanted as IMO empty request headers still means there's headers, just no actual value, as opposed to no request headers set yet. If yes, then I'll update the code to add deprecations to reflect on that.